numpy/testing/_private/utils.py

```diff
@@ -247,6 +247,7 @@ def build_err_msg(arrays, err_msg, header='Items are not equal:',
                   verbose=True, names=('ACTUAL', 'DESIRED'), precision=8):
     msg = ['\n' + header]
     if err_msg:
+        err_msg = str(err_msg)
```

Comment:
To be consistent with `assert` I'd suggest moving this line to before the `if err_msg: ...` evaluation.
Otherwise the likes of `err_msg=[]`, or any other object which evaluates to `False`. will fail to produce an error message.